src/components/signUpInOut/SignUpForm.js

```diff
@@ -0,0 +1,149 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import { useDispatch } from 'react-redux';
+import {
+  Form, Button, Row, Col, Modal,
+} from 'react-bootstrap';
+import { FaXmark } from 'react-icons/fa6';
+import { signUp } from '../../features/authSlice';
+
+const initialAddressData = {
+  street: '',
+  city: '',
+  state: '',
+  zip_code: '',
+};
+
+const initialFormData = {
+  name: '',
+  age: '',
+  email: '',
+  photo: '',
+  role: '',
+  password: '',
+  password_confirmation: '',
+  address: { ...initialAddressData },
+};
+
+const SignUpForm = ({ showSignUpModal, handleCloseSignUpModal, handleOpenSignUpModal }) => {
```

Comment:
- [ ] OPTIONAL: Regular user creation works for me but I was unable to create an admin user using the frontend.  If you came into issues during development that prevented you to implement this feature, kindly add an explanation to your PR description. 

Also, please add admin credentialls to your PR description, so people can test the admin functionality. 👍

Reply:
Thanks for the comment. noted